packages/flutter/lib/src/gestures/multitap.dart

```diff
@@ -587,3 +587,411 @@ class MultiTapGestureRecognizer extends GestureRecognizer {
+/// recognizes.
+///
+/// Each time a pointer contacts the screen, this recognizer will enter that
+/// gesture into the arena. This means that this recognizer will yield multiple
+/// winning entries in the arena for a single tap series as the series
+/// progresses.
+///
+/// If this recognizer loses the arena (either by declaring defeat or by
+/// another recognizer declaring victory) while the pointer is contacting the
+/// screen, it will fire [onSerialTapCancel], and [onSerialTapUp] will not
+/// be fired.
+///
+/// ### Button behavior
+///
+/// A tap series is defined to have the same buttons across all taps. If a tap
+/// with a different combination of buttons is delivered in the middle of a
+/// series, it will "steal" the series and begin a new series, starting the
+/// count over.
+///
+/// ### Interleaving tap behavior
+///
+/// A tap must be _completed_ in order for a subsequent tap to be considered
+/// "in the same series" as that tap. Thus, if tap A is in-progress (the down
+/// event has been received, but the corresponding up event has not yet been
+/// received), and tap B begins (another pointer contacts the screen), tap A
+/// will fire [onSerialTapCancel], and tap B will begin a new series (tap B's
+/// [SerialTapDownDetails.count] will be 1).
+///
+/// ### Relation to `TapGestureRecognizer` and `DoubleTapGestureRecognizer`
+///
+/// [SerialTapGestureRecognizer] fires [onSerialTapDown] and [onSerialTapUp]
+/// for every tap that it recognizes (passing the count in the details),
+/// regardless of whether that tap is a single-tap, double-tap, etc. This
+/// makes it especially useful when you want to respond to every tap in a
+/// series. Contrast this with [DoubleTapGestureRecognizer], which only fires
+/// if the user completes a double-tap, and [TapGestureRecognizer], which
+/// _doesn't_ fire if the recognizer is competing with a
+/// `DoubleTapGestureRecognizer`, and the user double-taps.
+///
+/// For example, consider a list item that should be _selected_ on the first
+/// tap and _cause an edit dialog to open_ on a double-tap. If you use both
+/// [TapGestureRecognizer] and [DoubleTapGestureRecognizer], there are a few
+/// problems: (1) if the user single-taps the list item, it will not select
+/// the list item until after enough time has passed to rule out a double-tap,
+/// and (2) if the user double-taps the list item, it will not select the list
+/// item at all. The solution is to use [SerialTapGestureRecognizer] and use
+/// the tap count to either select the list item or open the edit dialog.
+///
+/// ### When competing with `TapGestureRecognizer` and `DoubleTapGestureRecognizer`
+///
+/// Unlike [TapGestureRecognizer] and [DoubleTapGestureRecognizer],
+/// [SerialTapGestureRecognizer] aggressively declares victory when it detects
+/// a tap, so when it is competing with those gesture recognizers, it will beat
+/// them in the arena, regardless of which recognizer entered the arena first.
```

Comment:
Love the extensive documentation that explains in detail how this interacts with other recognizers!